src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,94 @@
+package parser;
+
+import command.AddCommand;
+import command.AllocateCommand;
+import command.CriteriaCommand;
+import command.DeleteCommand;
+import command.ExitCommand;
+import command.GenerateCommand;
+import command.HelpCommand;
+import command.ListCommand;
+import command.StatCommand;
+import command.UnknownCommand;
+import command.ViewQuotaCommand;
+import studentlist.StudentList;
+import ui.UI;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The Parser class is the main class responsible for parsing the user's input and executing the corresponding command.
+ */
+public class Parser {
+    private StudentList studentList;
+    private UI ui;
+
+    public Parser(StudentList studentList, UI ui) {
+        this.studentList = studentList;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the user's input and execute the corresponding command.
+     * 
+     * @return false if the user wants to exit the application, true otherwise.
+     */
+    public boolean parseInput(String input) {
+        String[] parts = input.split(" ");
+        switch (parts[0].toLowerCase()) {
+        case "add":
```

Comment:
You could consider extracting the commands into constant as well for easy usage in other places or change later on.